Testinfra fails if the "~/.ansible/cp" path doesn't exist

Hello,

After updating to release v9.0.0, we've noticed a regression. When the "~/.ansible/cp" path is missing, tests seem to fail with the following error:
```
self = <testinfra.backend.ssh.SshBackend object at 0x10199f3d0>
command = 'uname -s'

    def run_ssh(self, command: str) -> base.CommandResult:
        cmd, cmd_args = self._build_ssh_command(command)
        out = self.run_local(" ".join(cmd), *cmd_args)
        out.command = self.encode(command)
        if out.rc == 255:
            # ssh exits with the exit status of the remote command or with 255
            # if an error occurred.
>           raise RuntimeError(out)
E           RuntimeError: CommandResult(command=b'uname -s', exit_status=255, stdout=None, stderr=b"Warning: Permanently added '192.168.1.87' (ED25519) to the list of known hosts.\r\nunix_listener: cannot bind to path /Users/user/.ansible/cp/192.168.1.87-22-some-user.8yRLDJr64HeCjBh0: No such file or directory\r\n")

cmd        = ['ssh',
 '-F %s',
 '-o User=%s',
 '-o Port=%s',
 '-o ConnectTimeout=10',
 '-o ControlMaster=auto -o ControlPersist=60s',
 '-o ControlPath=~/.ansible/cp/%%h-%%p-%%r',
 '%s %s']
cmd_args   = ['../../../_molecule_resources/default/ssh_config',
 'some-user',
 '22',
 '192.168.1.87',
 'uname -s']
command    = 'uname -s'
out        = CommandResult(command=b'uname -s', exit_status=255, stdout=None, stderr=b"Warning: Permanently added '192.168.1.87' (ED25519) to the list of known hosts.\r\nunix_listener: cannot bind to path /Users/user/.ansible/cp/192.168.1.87-22-some-user.8yRLDJr64HeCjBh0: No such file or directory\r\n")
self       = <testinfra.backend.ssh.SshBackend object at 0x10199f3d0>
```

I believe the root cause might be associated with PR #713. 

It might be beneficial for Testinfra to verify if the "directory" exists prior to it's utilization.

Thank you for looking into this.

--- a/testinfra/backend/ssh.py
+++ b/testinfra/backend/ssh.py
@@ -27,7 +27,7 @@ class SshBackend(base.BaseBackend):
         ssh_config: Optional[str] = None,
         ssh_identity_file: Optional[str] = None,
         timeout: int = 10,
-        controlpath: str = "",
+        controlpath: Optional[str] = None,
         controlpersist: int = 60,
         ssh_extra_args: Optional[str] = None,
         *args: Any,

--- a/testinfra/utils/ansible_runner.py
+++ b/testinfra/utils/ansible_runner.py
@@ -187,13 +187,18 @@ def get_ansible_host(
 
     control_path = config.get("ssh_connection", "control_path", fallback="", raw=True)
     if control_path:
-        directory = config.get(
+        control_path_dir = config.get(
             "persistent_connection", "control_path_dir", fallback="~/.ansible/cp"
         )
-        control_path = control_path % ({"directory": directory})  # noqa: S001
-        # restore original "%%"
-        control_path = control_path.replace("%", "%%")
-        kwargs["controlpath"] = control_path
+        control_path_dir = os.path.expanduser(control_path_dir)
+        control_path_dir = os.path.normpath(control_path_dir)
+
+        if os.path.isdir(control_path_dir):
+            control_path = control_path % (  # noqa: S001
+                {"directory": control_path_dir}
+            )
+            control_path = control_path.replace("%", "%%")  # restore original "%%"
+            kwargs["controlpath"] = control_path
 
     spec = "{}://".format(connection)
 

